support/metrics/metricsTracker.go

```diff
@@ -286,3 +350,36 @@ func (mt *MetricsTracker) sendEvent(eventType string, eventProps interface{}) er
 	}
 	return nil
 }
+
+func mergeEventProps(commonProps, eventProps map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
commented in another pull request to avoid the pattern of input vars like `(v1, v2 int)` and instead use the more explicit version of `(v1 int, v2 int)`

Reply:
this function should not contain context of its usage -- here we are holding onto the context in the variable names with commonProps and eventProps.

better names would be `original` and `overrides`

Reply:
this is also a generic enough function that it can be moved to `utils/functions.go`